# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: gfx/layers/ipc/CompositorManagerParent.h
# Commit: 923dee1aab01
# Full Hash: 923dee1aab0128ea4ff60b876a8626af322084d8
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2023-12-15 05:00:55
# Description:
#   Bug 1869933 - Avoid allocating external image ID until ready for sharing with compositor process. r=gfx-reviewers,lsalzman
#   
#   There are some error paths where we could allocate an external image ID
#   for a SourceSurfaceSharedData before we are able to finish sharing it.
#   This patch makes it so that we defer allocating the ID until the last
# ==============================================================================

diff -r ef864a709ce1 -r 923dee1aab01 gfx/layers/ipc/CompositorManagerParent.h
--- a/gfx/layers/ipc/CompositorManagerParent.h	Thu Dec 14 23:17:37 2023 +0200
+++ b/gfx/layers/ipc/CompositorManagerParent.h	Thu Dec 14 20:08:59 2023 +0000
@@ -18,6 +18,11 @@
 #include "nsTArray.h"  // for AutoTArray
 
 namespace mozilla {
+
+namespace gfx {
+class SourceSurfaceSharedData;
+}
+
 namespace layers {
 
 class CompositorBridgeParent;
@@ -44,6 +49,9 @@
 
   static void WaitForSharedSurface(const wr::ExternalImageId& aId);
 
+  static void AddSharedSurface(const wr::ExternalImageId& aId,
+                               gfx::SourceSurfaceSharedData* aSurface);
+
   mozilla::ipc::IPCResult RecvAddSharedSurface(const wr::ExternalImageId& aId,
                                                SurfaceDescriptorShared&& aDesc);
   mozilla::ipc::IPCResult RecvRemoveSharedSurface(